### ISSUE TITLE
pg-strom on openSuSE?

### ISSUE THREAD

**Issue by camargocr** [heterodb/pg-strom#321]
Hi,
Please, is anyone using pg-strom on opensuse?
Thanks,
Carlos Camargo

**Comment by kaigai** [comment]
I usually run and develop on CentOS 7....

**Comment by camargocr** [comment]
I tried to run pg-Strom over openSuSE (version Leap 42.3), by myself (and risk!), although I know it is not supported on this OS. Apparently, there was no relevant issue on compilation and install, but it doesn't works (see attached log files). As you know, openSuSE is a relevant platform, so maybe I can contribute to help you port it, testing trials, if you agree about goals.
Best regards
[systemctl_error.log](https://github.com/heterodb/pg-strom/files/1597767/systemctl_error.log)
[make_warnings.log](https://github.com/heterodb/pg-strom/files/1597768/make_warnings.log)
[make.log](https://github.com/heterodb/pg-strom/files/1597769/make.log)

**Comment by kaigai** [comment]
It looks to me ```/usr/lib/postgresql96/bin/gpuinfo`` raised an error, during startup of PG-Strom.
Probably, CUDA setup is not correct.

Please check them
```
# echo /usr/local/cuda/lib64 > /etc/ld.so.conf.d/cuda-lib64.conf
# ldconfig
```
Then, run the command below.
```
$/usr/lib/postgresql96/bin/gpuinfo
```

**Comment by camargocr** [comment]
My config file:
```
# cat /etc/ld.so.conf.d/cuda-9-1.conf 
/usr/local/cuda-9.1/targets/x86_64-linux/lib
```
... and the output of gpuinfo:

```
# /usr/lib/postgresql96/bin/gpuinfo 
CUDA Runtime version: 9.1.0
NVIDIA Driver version: 387.26
Number of devices: 1
--------
Device Identifier: 0
Device Name: GeForce GTX 1080
Global memory size: 8110MB
Maximum number of threads per block: 1024
Maximum block dimension X: 1024
Maximum block dimension Y: 1024
Maximum block dimension Z: 64
Maximum grid dimension X: 2147483647
Maximum grid dimension Y: 65535
Maximum grid dimension Z: 65535
Maximum shared memory available per block in bytes: 48.0KB
Memory available on device for __constant__ variables in a CUDA C kernel in bytes: 64.0KB
Warp size in threads: 32
Maximum pitch in bytes allowed by memory copies: 2048.0MB
Maximum number of 32-bit registers available per block: 65536
Typical clock frequency in kilohertz: 1771.5MHz
Number of multiprocessors on device: 20
Specifies whether there is a run time limit on kernels: 1
Device is integrated with host memory: 0
Device can map host memory into CUDA address space: 1
Compute mode (See ::CUcomputemode for details): default
Device can possibly execute multiple kernels concurrently: true
Device has ECC support enabled: false
PCI bus ID of the device: 3
PCI device ID of the device: 0
Device is using TCC driver model: false
Peak memory clock frequency in kilohertz: 5.0GHz
Global memory bus width in bits: 256
Size of L2 cache in bytes: 2048.0KB
Maximum resident threads per multiprocessor: 2048
Number of asynchronous engines: 2
Device shares a unified address space with the host: true
PCI domain ID of the device: 0
Major compute capability version number: 6
Minor compute capability version number: 1
Device supports stream priorities: true
Device supports caching globals in L1: true
Device supports caching locals in L1: true
Maximum shared memory available per multiprocessor in bytes: 96.0KB
Maximum number of 32-bit registers available per multiprocessor: 65536
Device can allocate managed memory on this system: true
Device is on a multi-GPU board: false
Unique id for a group of devices on the same multi-GPU board: 0
Link between the device and the host supports native atomic operations: false
Ratio of single precision performance (in floating-point operations per second) to doubl
Device supports coherently accessing pageable memory without calling cudaHostRegister on
Device can coherently access managed memory concurrently with the CPU: true
Device supports compute preemption: true
Device can access host registered memory at the same virtual address as the CPU: false
```
CUDA setup seems ok, right?

**Comment by kaigai** [comment]
At least, it looks to me the first error you reported is solved.
GTX1080 is enough new (CC6.1; Pascal) to run.
